tests/driver/index.ts

```diff
@@ -0,0 +1,322 @@
+ */
+export interface TestResult {
+  success: boolean;
+  trunkRunResult: TrunkRunResult;
+  trunkVerb: TrunkVerb;
+  /** Absolute path to the target of the `trunk check`/`trunk fmt` run, if one was specified. */
+  targetPath?: string;
+  /** Attempt at parsing the outputJson and extracting only relevant information. */
+  landingState?: LandingState;
+}
+
+/**
+ * Configuration for when a TrunkDriver instance runs `setUp`.
+ */
+export interface SetupSettings {
+  /** Whether or not to run `git init` and attach a gitDriver. */
+  setupGit?: boolean;
+  /** Whether or not to create a new .trunk/trunk.yaml */
+  setupTrunk?: boolean;
+  /** Whether or not to launch the daemon on setup. */
+  launchDaemon?: boolean;
+}
+
+/**
+ * The primary means for setting up and running trunk commands in a test.
+ */
+export class TrunkDriver {
+  /** The name of the linter. If defined, enable the linter during setup. */
+  linter?: string;
+  /** Refers to the absolute path to the repo's test subdir inside a linter directory. */
+  testDir: string;
+  /** Created in /tmp during setup. */
+  sandboxPath?: string;
+  /** Created conditionally during setup. */
+  gitDriver?: git.SimpleGit;
+  /** What customization to use during setup. */
+  setupSettings: SetupSettings;
+
+  daemon?: ChildProcess;
+
+  constructor(testDir: string, setupSettings: SetupSettings, linter?: string) {
+    this.linter = linter;
+    this.testDir = testDir;
+    this.setupSettings = setupSettings;
+  }
+
+  /**
+   * Convert a TrunkRunResult into a full TestResult.
+   */
+  parseRunResult(
+    trunkRunResult: TrunkRunResult,
+    trunkVerb: TrunkVerb,
+    targetAbsPath?: string
+  ): TestResult {
+    return {
+      success: [0, 1].includes(trunkRunResult.exitCode) && trunkRunResult.stderr.length === 0,
+      trunkRunResult,
+      trunkVerb,
+      targetPath: targetAbsPath,
+      landingState: tryParseLandingState(this.testDir, trunkRunResult.outputJson),
+    };
+  }
+
+  /**
+   * Return the yaml result of parsing the output of `trunk config print` in the test sandbox.
+   */
+  getFullTrunkConfig = (): any => {
+    const printConfig = execSync(`${ARGS.cliPath ?? "trunk"} config print`, {
+      cwd: this.sandboxPath,
+    });
+    return YAML.parse(printConfig.toString());
+  };
+
+  /**
+   * Parse the result of 'getFullTrunkConfig' in the context of 'ARGS' to identify the desired linter version to enable.
+   */
+  extractLinterVersion = (): string => {
+    if (!ARGS.linterVersion || ARGS.linterVersion === "Latest") {
+      return "";
+    } else if (ARGS.linterVersion === "KnownGoodVersion") {
+      // trunk-ignore-begin(eslint/@typescript-eslint/no-unsafe-member-access,eslint/@typescript-eslint/no-unsafe-call)
+      return (
+        (this.getFullTrunkConfig().lint.definitions.find(
+          ({ name }: { name: string }) => name === this.linter
+        )?.known_good_version as string) ?? ""
```

Comment:
This type assertion shouldn't be necessary since you are coalescing to an empty string

Reply:
For some reason, eslint isn't smart enough to think so, so I'm leaving this in for now